addons/sale_timesheet/tests/test_reinvoice.py

```diff
@@ -283,3 +287,92 @@ def test_no_expense(self):
+          Confirm
+          -> Fails if there is nothing to invoice
+        """
+        product = self.env['product.product'].create({
+            'name': "Service delivered, create task in global project",
+            'standard_price': 30,
+            'list_price': 90,
+            'type': 'service',
+            'service_policy': 'delivered_timesheet',
+            'invoice_policy': 'delivery',
+            'default_code': 'SERV-DELI2',
+            'service_type': 'timesheet',
+            'service_tracking': 'task_global_project',
+            'project_id': self.project_global.id,
+            'taxes_id': False,
+            'property_account_income_id': self.account_sale.id,
+        })
+        today = Date.context_today(self.env.user)
+
+        # Creates a sales order for quantity 3
+        so_form = Form(self.env['sale.order'])
+        so_form.partner_id = self.env['res.partner'].create({'name': 'Toto'})
+        with so_form.order_line.new() as line:
+            line.product_id = product
+            line.product_uom_qty = 3.0
+        sale_order = so_form.save()
+        sale_order.action_confirm()
+
+        # "Deliver" 1 of 3
+        task = sale_order.tasks_ids
+        self.env['account.analytic.line'].create({
+            'name': 'Test Line',
+            'project_id': task.project_id.id,
+            'task_id': task.id,
+            'unit_amount': 1,
+            'employee_id': self.employee_user.id,
+            'company_id': self.company_data['company'].id,
+        })
+
+        context = {
+            "active_model": 'sale.order',
+            "active_ids": [sale_order.id],
+            "active_id": sale_order.id,
+            'open_invoices': True,
+        }
+        # Invoice the 1
+        wizard = self.env['sale.advance.payment.inv'].with_context(context).create({
+            'advance_payment_method': 'delivered'
+        })
+        invoice_dict = wizard.create_invoices()
+        # Confirm the invoice
+        invoice = self.env['account.move'].browse(invoice_dict['res_id'])
+        invoice.action_post()
+        # Refund the invoice
+        refund_invoice_wiz = self.env['account.move.reversal'].with_context(active_model="account.move", active_ids=[invoice.id]).create({
+            'reason': 'please reverse :c',
+            'refund_method': 'refund',
+            'date': today,
+        })
+        refund_invoice = self.env['account.move'].browse(refund_invoice_wiz.reverse_moves()['res_id'])
+        refund_invoice.action_post()
+        # reversing with action_reverse and then action_post does not reset the invoice_status to 'to invoice' in tests
+
+        # Recreate wizard to get the new invoices created
+        wizard = self.env['sale.advance.payment.inv'].with_context(context).create({
+            'advance_payment_method': 'delivered',
+            'date_start_invoice_timesheet': today - timedelta(days=7),
+            'date_end_invoice_timesheet': today + timedelta(days=7)
```

Comment:
Why do you specify dates here and not before? Is it required?

Reply:
The issue arise only when creating a return with a specific time period, the business flow we are following is when this condition https://github.com/odoo/odoo/blob/a392674a302b984b3a1e01a000717b4913403c38/addons/sale_timesheet/wizard/sale_make_invoice_advance.py#L40 is evaluated to True
it is only if the condition is True that we are going to look at the domain and try to extract what we need